pkg/proxy/stripeevent.go

```diff
@@ -31,25 +32,25 @@ func (e *StripeEvent) IsConnect() bool {
 
 // URLForEventID builds a full URL from a StripeEvent ID.
 func (e *StripeEvent) URLForEventID() string {
-	return fmt.Sprintf("%s/events/%s", BaseDashboardURL(e.Livemode, e.Account), e.ID)
+	return fmt.Sprintf("%s/events/%s", BaseDashboardURL(e.Livemode, e.Account, e.LoggedInAccountID), e.ID)
 }
 
 // URLForEventType builds a full URL from a StripeEvent Type.
 func (e *StripeEvent) URLForEventType() string {
-	return fmt.Sprintf("%s/events?type=%s", BaseDashboardURL(e.Livemode, e.Account), e.Type)
+	return fmt.Sprintf("%s/events?type=%s", BaseDashboardURL(e.Livemode, e.Account, e.LoggedInAccountID), e.Type)
 }
 
 // BaseDashboardURL constructs a dashboard URL with (optionally) the test and account values appended
-func BaseDashboardURL(livemode bool, account string) string {
+func BaseDashboardURL(livemode bool, connectedAccountID, loggedInAccountID string) string {
 	maybeTest := ""
 	if !livemode {
 		maybeTest = "/test"
 	}
 
-	maybeAccount := ""
-	if account != "" {
-		maybeAccount = fmt.Sprintf("/%s", account)
+	account := fmt.Sprintf("/%s", loggedInAccountID)
```

Comment:
To be defensive, could we check if `loggedInAccountID` is not empty? So that the URL is still valid.